src/tools/strings.ml

```diff
@@ -0,0 +1,120 @@
+(** Operations on lists of strings. This is module is used in order to avoid
+    concatenating (large) strings. Iterators are FIFO. *)
+
+(* List of "concatenated" strings, stored backwards. *)
+type t = string list
```

Comment:
We should make this API thread-safe and add mutex/mutexify behind the scene.

Reply:
Also, adding offset/length is indeed extra work but would be totally worth it.

Reply:
> Also, adding offset/length is indeed extra work but would be totally worth it.

If you want to do this, I would go for a "reimplementation" of the `String` module (say `OString`) which takes care of that transparently. This would mean that the module `OString` would look like
```
type t = string * int * int

let sub (s,o,l) o' l' = s (o+o') l'
```
and so on.